From 69963d4336b06285ecd4f2bc81ed9d7ee2678a63 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B6rn=20T=C3=B6pel?= <bjorn@rivosinc.com>
Date: Tue, 6 May 2025 07:16:01 +0000
Subject: [PATCH 2/7] selftests: mm/page_frag
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix build error for cross-builds. Make sure that correct kernel, and
make flags are picked up.

Signed-off-by: Björn Töpel <bjorn@rivosinc.com>
---
 tools/testing/selftests/mm/page_frag/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/testing/selftests/mm/page_frag/Makefile b/tools/testing/selftests/mm/page_frag/Makefile
index 8c8bb39ffa28..7e60c04dadf7 100644
--- a/tools/testing/selftests/mm/page_frag/Makefile
+++ b/tools/testing/selftests/mm/page_frag/Makefile
@@ -1,5 +1,5 @@
 PAGE_FRAG_TEST_DIR := $(realpath $(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
-KDIR ?= /lib/modules/$(shell uname -r)/build
+KDIR ?= $(abspath $(PAGE_FRAG_TEST_DIR)/../../../../..)
 
 ifeq ($(V),1)
 Q =
@@ -12,7 +12,7 @@ MODULES = page_frag_test.ko
 obj-m += page_frag_test.o
 
 all:
-	+$(Q)make -C $(KDIR) M=$(PAGE_FRAG_TEST_DIR) modules
+	+$(Q)$(MAKE) -C $(KDIR) M=$(PAGE_FRAG_TEST_DIR) modules
 
 clean:
-	+$(Q)make -C $(KDIR) M=$(PAGE_FRAG_TEST_DIR) clean
+	+$(Q)$(MAKE) -C $(KDIR) M=$(PAGE_FRAG_TEST_DIR) clean
-- 
2.48.1

